forge-game/src/main/java/forge/game/ability/effects/ChangeZoneEffect.java

```diff
@@ -1248,6 +1227,18 @@ else if (origin.contains(ZoneType.Hand) && player.isOpponentOf(decider)) {
             if (sa.hasParam("ShuffleChangedPile")) {
                 CardLists.shuffle(chosenCards);
             }
+
+            if (sa.hasParam("DestinationAlternative")) {
+                Map<ZoneType, Integer> map = handleAltDest(sa, source, destination, libraryPos, decider);
+                if (!map.containsKey(destination) || map.get(destination) != libraryPos) {
+                    for (final Map.Entry<ZoneType, Integer> e : map.entrySet()) {
```

Comment:
might be better to just use something like `Pair` so you don't loop for no reason